qutebrowser/components/braveadblock.py

```diff
@@ -0,0 +1,325 @@
+                # "file://". We handle those as a special case.
+                filename = url.toLocalFile()
+                if os.path.isdir(filename):
+                    for entry in os.scandir(filename):
+                        if entry.is_file():
+                            self._import_local(entry.path)
+                else:
+                    self._import_local(filename)
+            else:
+                download = downloads.download_temp(url)
+                self._in_progress.append(download)
+                download.finished.connect(
+                    functools.partial(self._on_download_finished, download)
+                )
+
+    def _import_local(self, filename: str) -> None:
+        """Adds the contents of a file to the blocklist.
+
+        Args:
+            filename: path to a local file to import.
+        """
+        try:
+            fileobj = open(filename, "rb")
+        except OSError as e:
+            message.error(
+                "adblock: Error while reading {}: {}".format(filename, e.strerror)
+            )
+            return
+        download = _FakeDownload(fileobj)
+        self._in_progress.append(download)
+        self._on_download_finished(download)
+
+    def _on_lists_downloaded(self) -> None:
+        """Install block lists after files have been downloaded."""
+        self._engine.serialize_to_file(str(self._cache_path))
+        logger.info("Block lists have been successfully imported")
+
+    def update_files(self) -> None:
+        """Update files when the config changed."""
+        if not config.val.content.blocking.adblock.lists:
+            try:
+                os.remove(self._cache_path)
+            except FileNotFoundError:
+                pass
+            except OSError as e:
+                logger.exception("Failed to adblock cache file: {}".format(e))
+
+    def _on_download_finished(self, download: downloads.TempDownload) -> None:
+        """Check if all downloads are finished and if so, trigger reading.
+
+        Arguments:
+            download: The finished download.
+        """
+        self._in_progress.remove(download)
+        if download.successful:
+            self._done_count += 1
+            assert not isinstance(download.fileobj, downloads.UnsupportedAttribute)
+            assert download.fileobj is not None
+            try:
+                download.fileobj.seek(0)
+                text = io.TextIOWrapper(download.fileobj, encoding="utf-8")
+                self._engine.add_filter_list(text.read())
+                text.close()
+            except UnicodeDecodeError:
+                message.info("Block list is not valid utf-8")
+            finally:
+                download.fileobj.close()
+        if not self._in_progress:
+            try:
+                self._on_lists_downloaded()
+            except OSError:
+                logger.exception("Failed to write host block list!")
+
+
+@cmdutils.register()
+def brave_adblock_update() -> None:
```

Comment:
I wonder if it makes sense to have a separate `:brave-adblock-update` command - I suppose the default `:adblock-update` could update both adblock lists at once?

Reply:
Yeah. I don't see a reason why we couldn't have one command to update both. I guess my only concern is how I'd implement it in code:

1. In which file would the `@cmdutils.register()` decorated function be located?
2. How would said function be able to access the `_ad_blocker` global variable?

Reply:
1. I suppose it could make sense to split it into yet another file, something like `components/adblockcommands.py` or whatever. I first thought it could be in the previously mentioned `blockutils.py`, but that'd likely lead to a circular import as both `braveadblock.py` and `adblock.py` need to import from `blockutils.py`; but `blockutils.py` would also need to import from those.
2. If the global variable is renamed to `ad_blocker` to be "public", it should then be able to do:

```python
from qutebrowser.components import braveadblock
braveadblock.ad_blocker
```